mappings/net/minecraft/world/ModifiableWorld.mapping

```diff
@@ -12,6 +12,27 @@ CLASS net/minecraft/class_1945 net/minecraft/world/ModifiableWorld
 		ARG 2 drop
 		ARG 3 breakingEntity
 	METHOD method_8652 setBlockState (Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;I)Z
+		COMMENT Updates the blockstate at a position within this world calling all appropriate callbacks and event handlers.
+		COMMENT <p>
+		COMMENT When called on the server, the new blockstate is stored and propagated to clients and listeners as dictated
+		COMMENT by the supplied flags. Note that calling this on the client will update the world locally, but may not see the
+		COMMENT change persisted across loads. It's recommended to check whether running on the client or server before
+		COMMENT interacting with the world in this way.
+		COMMENT <p>
+		COMMENT Accepted values of the flags are a bitset combination of the following:
+		COMMENT <p>
+		COMMENT 0b0000000 // 0 - DEFAULT/NONE - nothing is notified. The state is merely stored without any visual changes
+		COMMENT 0b0000001 // 1 - PROPAGATE_CHANGE - Propagates a change event to surrounding blocks.
+		COMMENT 0b0000010 // 2 - NOTIFY_LISTENERS - Notifies listeners and clients who need to react when the block changes
+		COMMENT 0b0000100 // 4 - NO_REDRAW - Used in conjunction with NOTIFY_LISTENERS to suppress the render pass on clients.
+		COMMENT 0b0001000 // 8 - REDRAW_ON_MAIN_THREAD - Forces a synchronous redraw on clients.
+		COMMENT 0b0010000 // 16 - FORCE_STATE - Bypass virtual blockstate changes and forces the passed state to be stored as-is.
+		COMMENT 0b0100000 // 32 - SKIP_DROPS - Prevents the previous block (container) from dropping items when destroyed.
+		COMMENT 0b1000000 // 64 - IS_MECHANICAL_UPDATE - Signals that this is a mechanical update, usually caused by pistons moving blocks.
+		COMMENT </p>
```

Comment:
I know we dont have any convention yet, is this needed here?

Reply:
What's the issue? The whole point was to document the flags and it would have been too long to put into the @param for the `flags` parameter. (iirc can't have more than one line)

Reply:
Or do you mean the `</p>`?

That's there because otherwise there is no blank line between the end of the description and the first param.

Reply:
Humm, I think I add that in loom:

https://github.com/FabricMC/fabric-loom/blob/dev/0.2.7/src/main/java/net/fabricmc/loom/task/fernflower/TinyJavadocProvider.java#L104

I think engima should do the same.

Reply:
Ah. Okay.

I tried manually adding it, but enogma stripped the blank line.

Reply:
damned enogma

Reply:
Do we want to allow blank lines? I think it may be better than <p> ?

Reply:
In eclipse line breaks and blank lines don't really render that well (hence the HTML markup), so I don't know.

Reply:
If that is the case then I should change it to use <p> in loom.

Reply:
![image](https://user-images.githubusercontent.com/6429283/70626241-bcce5900-1c2c-11ea-8996-4614dc21f72e.png)
I think actually I should convert the newlines here into proper markup. I always forget that it doesn't display well.